### PR TITLE
Return multiple modules in `parent_module` feature

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: crates/ide/src/parent_module.rs [hunk 1]
```diff
@@ -18,8 +19,7 @@ use crate::NavigationTarget;
 // | VS Code | **Rust Analyzer: Locate parent module**
 // |===
 
-/// This returns `Vec` because a module may be included from several places. We
-/// don't handle this case yet though, so the Vec has length at most one.
+/// This returns `Vec` because a module may be included from several places.
```
Comment (lnicola) [inline crates/ide/src/parent_module.rs:22]: :sob:

**Comment by lnicola** [comment]
changelog internal prepare for returning parents in the "Locate parent module" command.

File: crates/hir/src/semantics.rs [hunk 2]
```diff
@@ -259,6 +259,10 @@ impl<'db, DB: HirDatabase> Semantics<'db, DB> {
     }
 
     pub fn to_module_def(&self, file: FileId) -> Option<Module> {
+        self.imp.to_module_def(file).next()
+    }
+
+    pub fn to_module_defs(&self, file: FileId) -> impl Iterator<Item = Module> {
```
Comment (matklad) [inline crates/hir/src/semantics.rs:265]: In the future, we should get rid of the first function. It's problematic in that it's not obvious, at the call site, that more than one result can be returned.  Although, just returening a vec in call cases seems wrong as well. Rather, the caller probably wants to pass in some kind of predicate object, which is used to select the best possible module. 

So, something like `fn guess_module_def(file: FiledId, node: &SyntaxNode)` might make sens.

**Review by matklad — COMMENTED** [review]
LGTM!

**Comment by Veykril** [comment]
bors r=matklad

File: crates/hir/src/semantics.rs (hunk 2 above)
Comment (Veykril) [reply crates/hir/src/semantics.rs:265]: Ye was thinking about that as well a bit, getting the predicate/heuristic right seems difficult

**Pull request merged**